### PR TITLE
Moving Wyre transaction from iFrame to pop-up window

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 1]
```diff
@@ -75,6 +75,10 @@ export const Onboarding = (props: OnboardingProps) => {
                   />
                 </ModalWrapper>}
             />
+            <Route
+              path='/waitingForWyre'
+            >
+            </Route>
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx:81]:
```suggestion
```

File: universal-login-react/src/ui/commons/WaitingForWyre.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  onBack: () => void;
+}
+
+export const WaitingForWyre = ({onBack}: WaitingForWyreProps) => {
+  return <div className={useClassFor('waitingfortransaction')}>
+    <WaitingFor
+      action = 'Waiting for Wyre transaction'
+      description = 'We have opened the Wyre in a new window. If you complete a transaction, this window will update automatically. If you want to change provider, you can click the button and go back to selection'
+    />
+    <button onClick={onBack} className="unilogin-component-pay-btn unilogin-theme-unilogin">
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/commons/WaitingForWyre.tsx:15]: use helper functions e.g. `useThemedComponent()`

**Comment by JustCaptcha** [comment]
image

File: universal-login-react/src/ui/TopUp/OnRamp/Wyre.tsx [hunk 3]
```diff
@@ -1,56 +1,38 @@
...
-    } catch {
-      onError?.();
-    }
+export const Wyre = ({address, currency, config, onBack, isDeployed}: WyreProps) => {
+  function openPopUp() {
+    const url = getWyreUrl(address, currency, config);
+    const width = 470;
+    const height = 630;
+    const top = window.outerHeight / 2 + window.screenY - (height / 2);
+    const left = window.outerWidth / 2 + window.screenX - (width / 2);
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/TopUp/OnRamp/Wyre.tsx:19]: Extract to a helper function

File: universal-login-react/src/ui/commons/getWindowCenterForPopUp.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const getWindowCenterForPopUp = (width: number, height: number) => {
+  const top = window.outerHeight / 2 + window.screenY - (height / 2);
+  const left = window.outerWidth / 2 + window.screenX - (width / 2);
+  return [top, left];
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/commons/getWindowCenterForPopUp.ts:4]: I would consider returning `{top, left}`

File: universal-login-react/src/ui/TopUp/OnRamp/Wyre.tsx [hunk 5]
```diff
@@ -1,56 +1,38 @@
...
   }
 
-  return (
-    <iframe
-      src={url}
-      onLoad={onLoad}
-    />
-  );
+  openPopUp();
+  return isDeployed ? <>{onBack()}</> : <WaitingForWyre onBack={onBack}/>;
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/TopUp/OnRamp/Wyre.tsx:26]: Can we make it different way? It looks very strange. `<>{onBack()}</>`

**Pull request merged**